### DISCUSSION TITLE
Im newbe with lang server and i dont know how to use

### DISCUSSION THREAD

**Discussion by lucas-gio** [fwcd/kotlin-language-server#386] [Ideas]
Is my case. I have an download section with server and grammar. Ok. ¿and?
¿How to install? My first impression was: I must compile this in order to use.
The readme is focused in server's developers, but not in server's users.
¿I need to unzip this in an folder that i must add later in path env?
¿How to use?

In server folder i have an bat. I can use to start this program, but... ¿?

What I'm trying to say is that maybe a step by step is missing for those who just see this server and want to use it.
Thanks for read my discussion!!

**Comment by themkat** [comment]
> The readme is focused in server's developers, but not in server's users.

Where is this information for the server's developers??? 😱 Actually having that there would have been super useful when I first started contributing 😛 Sorry for messing with you on this one, now to the serious reply....

> In server folder i have an bat. I can use to start this program, but... ¿?

First of all, what editor are you planning to use the server with? The server only provides IDE like functionality as a backend/server application, but you still need an editor where these features should be used from. The language server is not a standalone application in that regard. We have a list of [various editors and their config linked to in the README](https://github.com/fwcd/kotlin-language-server/blob/main/EDITORS.md). After you have picked an editor you want to give IDE-like super powers to, then you will need to configure that editor to use the language server application.

Have summarised it for the editors I'm most familiar with below 🙂 Let me know if you need any additional information. 


### VSCode
If you are using VSCode, then [vscode-kotlin](https://github.com/fwcd/vscode-kotlin) is probably your best bet. It can even be downloaded [from the VSCode Marketplace](https://marketplace.visualstudio.com/items?itemName=fwcd.kotlin). When you start it from the first time, it will download the language server and set up everything for you 🙂 If you prefer to provide the language server yourself (by giving it an executable path), you can change that in the Kotlin extension settings. (It also works with [Kotlin-debug-adapter](https://github.com/fwcd/kotlin-debug-adapter), if you need that for simple debugging.).


### Emacs
If you enjoy using Emacs, like myself, you can simply use the `lsp-mode` package. Then run the `lsp` command inside a Kotlin source file and import the root of the project. Like the VSCode extension, `lsp-mode` will download the server for you if you do not already have it installed. I have also written [a blog post with some tips and tricks to work with Kotlin in Emacs](https://themkat.net/2021/11/03/kotlin_in_emacs.html). It includes basic Emacs lisp config. 


### Others (Vim, Sublime etc.)
For Vim, [the EDITOR.md document](https://github.com/fwcd/kotlin-language-server/blob/main/EDITORS.md#vim) describes installation quite well.

[Sublime seems to have an LSP extension that you can try](https://github.com/sublimelsp/LSP).
